src/components/Item.js

```diff
@@ -42,21 +57,41 @@ const Item = ({ item, handleChange, deleteItem }) => {
   };
 
   return (
-    <li className={`list-item ${addNextPurchaseStyling(item)}`}>
-      <input
-        className="checkbox"
-        type="checkbox"
-        checked={checked}
-        onChange={e => handleChange(e, item)}
-        id={item.id}
-      />
-      <label htmlFor={item.id} className={className}>
-        {item.name} - next purchase in {item.nextPurchase} days
-      </label>
-      <button className="delete" onClick={() => deleteItem(item.id)}>
-        Delete
-      </button>
-    </li>
+    <>
+      <li className={`list-item ${addNextPurchaseStyling(item)}`}>
+        <label htmlFor={item.id}>
+          <input
+            className="checkbox"
+            type="checkbox"
+            checked={checked}
+            onChange={e => handleChange(e, item)}
+            id={item.id}
+          />
+        </label>
+
+        <span className={className} onClick={handleModalChange}>
+          {item.name}
+        </span>
+
+        <button className="delete" onClick={() => deleteItem(item.id)}>
+          Delete
+        </button>
+      </li>
+      {showItemInfo && (
+        <Modal setDisplay={setShowItemInfo}>
+          <h2>{item.name}</h2>
+          <ul>
+            <li>Last purchased on {lastPurchasedDate}</li>
+            <li>Next purchase in {item.nextPurchase} days</li>
+            <li>
+              {item.numberOfPurchases > 1
```

Comment:
👏👏👏 I went looking for whether you handled both singular and plural cases and was impressed that you thought of it!